File: p5js/js/sf2_functions.js
"use strict";

let sketch_2 = function(p) {

	let x = 0;
	let y = 100;
	let s = 50;

	p.setup = function() {
	  p.createCanvas(200, 200);
	}

	p.draw = function() {
	  p.background(127);
	  p.drawCircle();
	  p.text("(click)", 5, p.height-5);
	}

	p.mousePressed = function() {
		x = 0;
	}

	p.drawCircle = function() {
	  p.ellipse(x, y, s, s);
	  x += 0.5;
	}

}

let p5_2 = new p5(sketch_2, "canvas2");

File: p5py/sf2_functions.py
from p5 import *

x = 0
y = 100
s = 50

def setup():
    size(200, 200)

def draw():
    background(127)
    drawCircle()

def mouse_pressed():
    global x
    x = 0
    
def drawCircle():
    global x, y, s
    ellipse(x, y, s, s)
    x += 0.5
    
run()